docs/FAQ.md

```diff
@@ -0,0 +1,19 @@
+# FAQ for s2n-tls
+
+### Why is the ClientHello version reported as TLS 1.2 but the handshake itself negotiated TLS 1.3?
+The TLS 1.3 spec uses the TLS 1.2 version in ClientHello and on record layers for compatibility with middleboxes and poor server implementations. There is a separate ClientHello extension which lists the actual versions supported.
+
+### Why is my TLS handshake failing on validating my peer’s certificate? I gave my s2n endpoint a root certificate that verifies the peer’s certificate.
+You may need to validate the peer’s hostname by implementing the s2n_verify_host_fn callback. The default behavior may need to be altered for your usecase.
```

Comment:
Maybe make this more generally "Why is negotiation failing on validating my peer's certificate" and include how to set up the trust store as the first step?